packages/eslint-plugin/src/rules/use-component-view-encapsulation.ts

```diff
@@ -2,52 +2,66 @@ import type { TSESTree } from '@typescript-eslint/experimental-utils';
 import { createESLintRule } from '../utils/create-eslint-rule';
 import { COMPONENT_CLASS_DECORATOR } from '../utils/selectors';
 import {
-  getDecoratorPropertyValue,
-  isIdentifier,
-  isMemberExpression,
+  getImportDeclarations,
+  getImportRemoveFix,
+  getNodeToCommaRemoveFix,
 } from '../utils/utils';
 
 type Options = [];
-export type MessageIds = 'useComponentViewEncapsulation';
+export type MessageIds =
+  | 'useComponentViewEncapsulation'
+  | 'suggestRemoveViewEncapsulationNone';
 export const RULE_NAME = 'use-component-view-encapsulation';
-
-const NONE = 'None';
+const VIEW_ENCAPSULATION_NONE = 'ViewEncapsulation.None';
 
 export default createESLintRule<Options, MessageIds>({
   name: RULE_NAME,
   meta: {
     type: 'suggestion',
     docs: {
-      description: `Disallows using ViewEncapsulation.${NONE}`,
+      description: `Disallows using \`${VIEW_ENCAPSULATION_NONE}\``,
       category: 'Best Practices',
       recommended: false,
     },
     schema: [],
     messages: {
-      useComponentViewEncapsulation: `Using ViewEncapsulation.${NONE} makes your styles global, which may have an unintended effect`,
+      useComponentViewEncapsulation: `Using \`${VIEW_ENCAPSULATION_NONE}\` makes your styles global, which may have an unintended effect`,
+      suggestRemoveViewEncapsulationNone: `Remove \`${VIEW_ENCAPSULATION_NONE}\``,
     },
   },
   defaultOptions: [],
   create(context) {
-    return {
-      [COMPONENT_CLASS_DECORATOR](node: TSESTree.Decorator) {
-        const encapsulationExpression = getDecoratorPropertyValue(
-          node,
-          'encapsulation',
-        );
-
-        if (
-          !encapsulationExpression ||
-          (isMemberExpression(encapsulationExpression) &&
-            isIdentifier(encapsulationExpression.property) &&
-            encapsulationExpression.property.name !== NONE)
-        ) {
-          return;
-        }
+    const sourceCode = context.getSourceCode();
 
+    return {
+      [`${COMPONENT_CLASS_DECORATOR} Property[key.name=encapsulation][value.object.name='ViewEncapsulation'][value.property.name='None']`](
+        node: TSESTree.Property,
+      ) {
         context.report({
-          node: encapsulationExpression,
+          node: node.value,
           messageId: 'useComponentViewEncapsulation',
+          suggest: [
+            {
+              messageId: 'suggestRemoveViewEncapsulationNone',
+              fix: (fixer) => {
+                const importDeclarations = getImportDeclarations(
+                  node,
+                  '@angular/core',
+                );
+
+                return [
+                  getNodeToCommaRemoveFix(sourceCode, node, fixer),
+                ].concat(
```

Comment:
Why is concat being used here? Why not a simple array including the return values of both functions?

Reply:
I used `.concat` because the return type of the`getImportRemoveFix` function can be either a single item or an `array`. See #518.